pkg/operation/botanist/etcd.go

```diff
@@ -56,6 +57,11 @@ func (b *Botanist) DefaultEtcd(role string, class etcd.Class) (etcd.Interface, e
 		replicas = pointer.Int32(getReplicas(b.Shoot.GetInfo()))
 	}
 
+	k8sGTE121, err := versionutils.CompareVersions(b.ShootVersion(), ">=", "1.21")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Instead of comparing the Kubernetes version here and passing the bool value to `NewEtcd`, we should pass Kubernetes version to `NewEtcd` and make the required comparisons in `etcd` package.

Reply:
Why?

Reply:
Suppose there comes another scenario in future that depends on some other Kubernetes version. We can't pass another bool variable for that as this can keep growing.

Reply:
Because of supportability reason. If tomorrow we need a constraint >= 1.25, what should we do - add yet another bool named `k8sGTE125`?

Reply:
Are you aware of such case as of today?
If there is in future, this code can be refactored then and pass the version instead of the boolean check.